src/crimson/os/seastore/onode_manager/staged-fltree/fltree_onode_manager.cc

```diff
@@ -176,7 +182,12 @@ FLTreeOnodeManager::get_or_create_onode(
 	cursor.value());
     if (created) {
       DEBUGT("created onode for entry for {}", trans, hoid);
-      onode->create_default_layout(trans);
+      auto key = hoid.hobj.get_key();
+      if (key.empty()) {
+	onode->create_default_layout(trans, hoid.hobj.oid.name);
+      } else {
+	onode->create_default_layout(trans, key);
+      }
```

Comment:
Is it correct that the onode value needs to store the oid only if the locator is valid (!key.empty()), otherwise, there is no need to store or even reserve the value space?

Reply:
If there's other ways to determine whether the "oid" field in the onode tree key is actually oid or locator key, yes.

Reply:
> Is it correct that the onode value needs to store the oid only if the locator is valid (!key.empty()), otherwise, there is no need to store or even reserve the value space?

Yes, but I think the size of onode_layout_t has to be a constant value, is this right?

Reply:
I suppose the reason why we were not aware of the issue because its rare that `key.empty()`, so we mostly don't need the space to store the oid in the value.

> Yes, but I think the size of onode_layout_t has to be a constant value, is this right?

I think we might be able to store it in `xattr` omap with a special internal-only label. So oid becomes really optional. What do you think?

Reply:
will do.

Reply:
> I think we might be able to store it in `xattr` omap with a special internal-only label. So oid becomes really optional. What do you think?

I think we need a way to determine whether the "oid" field in the key is actually the oid or locator key, at least when listing objects, so that we can reconstruct the `ghobject_t` correctly. If we put oid in xattr, we'll need to search the omap tree for each of the objects that are listed. This seems to be a potential performance issue.